Storage/src/StreamWrapper.php

```diff
@@ -127,6 +127,24 @@ public function stream_set_option()
         return false;
     }
 
+    /**
+     * This is called when touch is used on a stream. See:
+     * https://www.php.net/manual/en/streamwrapper.stream-metadata.php
+     */
+    public function stream_metadata($path, $option, $value)
+    {
+        if ($option == STREAM_META_TOUCH) {
+            $this->openPath($path);
+            $this->bucket->upload('', [
```

Comment:
As per my understanding, it is for adding `metadata`, why are we creating file in this function?

Reply:
We need to take care of the case were file is already present, then we just need to update the file instead of trying to create it again.

Reply:
Fixed the second comment.
A file needs to be created when touch is called.